libs/cardano-ledger-api/src/Cardano/Ledger/Api/State/Query.hs

```diff
@@ -185,6 +190,41 @@ queryDRepStakeDistr nes creds
   where
     distr = psDRepDistr . fst $ finishedPulserState nes
 
+-- | Query the stake distribution of the registered DReps. This does not
+-- include the @AlwaysAbstain@ and @NoConfidence@ DReps.
+queryRegisteredDRepStakeDistr ::
+  ConwayEraGov era =>
+  NewEpochState era ->
+  -- | Specify DRep Ids whose stake distribution should be returned. When this set is
+  -- empty, distributions for all of the registered DReps will be returned.
+  Set (Credential 'DRepRole (EraCrypto era)) ->
+  Map (Credential 'DRepRole (EraCrypto era)) Coin
+queryRegisteredDRepStakeDistr nes creds
+  | null creds = distr
+  | otherwise = distr `Map.restrictKeys` creds
+  where
+    distr = Map.map fromCompact $ Map.foldlWithKey' computeDistr mempty registeredDReps
+    registeredDReps = nes ^. nesEsL . esLStateL . lsCertStateL . certVStateL . vsDRepsL
+    computeDistr distrAcc stakeCred (DRepState {..}) =
+      Map.insert
+        stakeCred
+        (addCompact (fromJust $ toCompact drepDeposit) $ totalDelegations drepDelegs)
+        distrAcc
+    totalDelegations =
+      Set.foldl'
+        (\acc c -> addCompact acc $ stakeAndDeposits c incrStake pDeposits)
+        (CompactCoin 0)
+    incrStake = nes ^. nesEsL . epochStateIncrStakeDistrL
+    pDeposits = proposalsDeposits $ nes ^. newEpochStateGovStateL . proposalsGovStateL
+    stakeAndDeposits stakeCred stakeDistr proposalDeposits =
```

Comment:
Slight simplifications:
* `CompactCoin` has a `Monoid` instance and `Set` a `Foldable`, so we can use `foldMap`
* local bindings do not need to be passed as function arguments, they are already available ins scope
* We can avoid `Map.map fromCompact`, by doing the conversion here instead.
```suggestion
      fromCompact . foldMap stakeAndDeposits
    stakeDistr = nes ^. nesEsL . epochStateIncrStakeDistrL
    proposalDeposits = proposalsDeposits $ nes ^. newEpochStateGovStateL . proposalsGovStateL
    stakeAndDeposits stakeCred =
```